scripts/endpoints-ruleset/compress.js

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env node
+
+const fs = require("fs");
+const path = require("path");
+
+const RemoveWhitespace = require("./compression-algorithms/RemoveWhitespace");
+const PatternDetection = require("./compression-algorithms/PatternDetection");
+
+/**
+ * Run compression on ruleset objects for SDK clients.
+ */
+const main = () => {
+  const root = path.join(__dirname, "..", "..");
+  const clientsFolder = path.join(root, "clients");
+  const modelsFolder = path.join(root, "codegen", "sdk-codegen", "aws-models");
```

Comment:
This performs compression on all clients, which is time consuming for client generation.
Also, the clients may be generated from any glob, although we primarily use models from `codegen/sdk-codegen/aws-models`

The compression needs to be performed only on generated clients.
May be pass codegen output path to it, and only process the clients generated in that directory?

For example, the value in `CODE_GEN_SDK_OUTPUT_DIR` can be passed. It can be called after prettifyCode https://github.com/aws/aws-sdk-js-v3/blob/b03bc8f09c20173b39be840c83fa6fcc9e25adf8/scripts/generate-clients/index.js#L80

Reply:
Instead of reading the ruleset from model, the compression algorithm can attempt reading it from `ruleset.ts` instead. But that would be time consuming.

I'll resolving this conversation as of now. This is a minor issue, and can be taken up as a fast-follow.